pgcontents/query.py

```diff
@@ -798,26 +807,33 @@ def _generate_notebooks(table, timestamp_column,
 
     # Decrypt each notebook and yield the result.
     for nb_row in result:
-        # The decrypt function depends on the user
-        user_id = nb_row['user_id']
-        decrypt_func = crypto_factory(user_id).decrypt
-
-        nb_dict = to_dict_with_content(table.c, nb_row, decrypt_func)
-        if table is files:
-            # Correct for files schema differing somewhat from checkpoints.
-            nb_dict['path'] = nb_dict['parent_name'] + nb_dict['name']
-            nb_dict['last_modified'] = nb_dict['created_at']
-
-        # For 'content', we use `reads_base64` directly. If the db content
-        # format is changed from base64, the decoding should be changed
-        # here as well.
-        yield {
-            'id': nb_dict['id'],
-            'user_id': user_id,
-            'path': to_api_path(nb_dict['path']),
-            'last_modified': nb_dict['last_modified'],
-            'content': reads_base64(nb_dict['content']),
-        }
+        try:
+            # The decrypt function depends on the user
+            user_id = nb_row['user_id']
+            decrypt_func = crypto_factory(user_id).decrypt
+
+            nb_dict = to_dict_with_content(table.c, nb_row, decrypt_func)
```

Comment:
If the `CorruptedFile` error can occur from the decrypt call, could you isolate that to be the only thing called within the `try` block? That way it makes it easier to follow what it is exactly that can raise an exception.